crates/sui-types/src/crypto.rs

```diff
@@ -145,6 +191,68 @@ impl AsRef<[u8]> for PublicKey {
     }
 }
 
+impl From<Ed25519PublicKey> for PublicKey {
+    fn from(key: Ed25519PublicKey) -> Self {
+        PublicKey::Ed25519KeyPair(key)
+    }
+}
+
+impl From<Secp256k1PublicKey> for PublicKey {
+    fn from(key: Secp256k1PublicKey) -> Self {
+        PublicKey::Secp256k1KeyPair(key)
+    }
+}
+
+impl EncodeDecodeBase64 for PublicKey {
+    fn encode_base64(&self) -> String {
+        let mut bytes: Vec<u8> = Vec::new();
+        bytes.extend_from_slice(&[self.flag()]);
+        bytes.extend_from_slice(self.as_ref());
+        base64ct::Base64::encode_string(&bytes[..])
+    }
+
+    fn decode_base64(value: &str) -> Result<Self, eyre::Report> {
+        let bytes =
+            base64ct::Base64::decode_vec(value).map_err(|e| eyre::eyre!("{}", e.to_string()))?;
+        match bytes.first() {
+            Some(x) => {
+                if x == &<Ed25519PublicKey as SuiPublicKey>::SIGNATURE_SCHEME.flag() {
+                    let pk = Ed25519PublicKey::from_bytes(&bytes[1..])?;
+                    Ok(PublicKey::Ed25519KeyPair(pk))
+                } else if x == &<Secp256k1PublicKey as SuiPublicKey>::SIGNATURE_SCHEME.flag() {
+                    let pk = Secp256k1PublicKey::from_bytes(&bytes[1..])?;
+                    Ok(PublicKey::Secp256k1KeyPair(pk))
+                } else {
+                    Err(eyre::eyre!("Invalid flag byte"))
+                }
+            }
+            _ => Err(eyre::eyre!("Invalid bytes")),
```

Comment:
In a future PR, we need to have a well defined tutorial on what do we need to change when adding a new flag (scheme).